### DISCUSSION TITLE
Is it possible to create IFactoriy<T>, IFactoriy<T1, T> automatically?

### DISCUSSION THREAD

**Discussion by ivmazurenko** [YairHalberstadt/stronginject#196] [Q&A]
Hello, I have autofac-like container in my project. There is a lot of code (and tests) presented that depends on IFactories like following:

```c#
    public class Factory<TResult> : IFactory<TResult>  {
        private readonly IComponentContext _context;
        public Factory(IComponentContext context) =>  _context = context;

        public TResult Create()
        {
            return _context.Resolve<TResult>();
        }
    }
```

So, I am expecting to use Stronginject same way. This is obfuscated example, where I use a lot of factories:

```c#
    public class SomeClass {
        public SomeClass(
            IFactory<IService1> service1Factory,
            IFactory<IService2> service2Factory,
            IFactory<IService3> service3Factory,
            IFactory<IService4> service4Factory,
            IFactory<IService5> service5Factory,
            IFactory<IService6> service6Factory,
            IFactory<IDependancy1, IService7> service7Factory,
            IFactory<IDependancy2, IDependancy3 IService8> service8Factory)
        {
            // ...
        }
    }
```

As I see, IFactory in StrongInject required to be implemented manually. But in my case it works automatically.

I do not want to rewrite all of these code. I just want replace my container with Stronginject in order to check its performance.  

How can I adapt Stronginject container to my code?

**Comment by YairHalberstadt** [comment]
You can write a generic factory method:

```csharp
[Factory] IFactory<T> CreateFactory<T>(){
}
```

**Reply by ivmazurenko** [reply]
It is empty? I can not understand.

Also, I tried something like that. It works for me:

    [Register(typeof(Fac<>), Scope.InstancePerDependency, typeof(Core.Container.IFactory<>))]

    public sealed class Fac<T> : IFactory<T>
    {
        private readonly Func<T> _func;
        public Fac(Func<T> func) => _func = func;
        public T Create() => _func();
    }

But i am still can not understand, how to put dependency services into created with IFactory instance

**Reply by YairHalberstadt** [reply]
> But i am still can not understand, how to put dependency services into created with IFactory instance

What do you mean by that?

**Reply by ivmazurenko** [reply]
Nothing, it works
